core/vm/runtime/runtime.hpp

```diff
@@ -141,6 +142,59 @@ namespace fc::vm::runtime {
     /// Update actor state CID
     virtual outcome::result<void> commit(const ActorSubstateCID &new_state) = 0;
 
+    /// Resolve address to id-address
+    virtual outcome::result<Address> resolveAddress(const Address &address) = 0;
+
+    /// Verify PoSt
+    virtual outcome::result<bool> verifyPoSt(uint64_t sector_size,
+                                             const PoStVerifyInfo &info) = 0;
+
+    /// Verify seal
+    virtual outcome::result<bool> verifySeal(uint64_t sector_size,
+                                             const SealVerifyInfo &info) = 0;
+
+    /// Send funds
+    inline auto sendFunds(const Address &to, BigInt value) {
+      // kSendMethodNumber circular dependency
+      return send(to, {0}, {}, value);
+    }
+
+    /// Send with typed result R
+    template <typename R>
+    outcome::result<R> sendR(Address to_address,
```

Comment:
I don't like the sendPR methods. We have encodeActorParams/decodeActorParams functions, and these methods seems to be an overkill. Names are a bit tricky (sendPR). For me these methods do not make code more readable and simpler.

Reply:
encodeActorParams and decoding result would use two extra OUTCOME_TRY, with need for 2 extra variable names.
Hiding encoding and decoding into templates seems better to me